### PR TITLE
Harden SPDX checks and correct 50 Jinja license headers

### PR DESCRIPTION
**Pull request by saturley-hall**
## Why and what changed

OSRB's follow-up review of the non-dataset portion of `release/0.12.0` found four remaining documentation and SPDX issues. This change hardens the repository check first, fixes all 50 malformed Jinja SPDX identifiers, adds the literal DCO sign-off command, explains the wheel-local legal-file copies from the root README, and aligns the NVIDIA-authored replay-parity reference with the repository's Apache-2.0 license.

## Review map

- Risk level: low
- Start with: `scripts/check_copyright.py`, then the seven commits in order
- Public or serialized contract changed: no
- Compatibility or rollback concern: the template changes are confined to non-rendered Jinja comments and retain whitespace trimming; rollback is a normal revert

## Evidence

- Tests: `339 passed, 4 skipped` in `python/aisimulate/tests/unit/generator`; `35 passed` in `tests/test_source_compliance.py`
- Fast CI: passed at `c730a22c78074d66550cec35f757f7e91d9e0c60` ([run](https://github.com/ai-dynamo/aisimulate/actions/runs/34308646863))
- Full CI: passed at `c730a22c78074d66550cec35f757f7e91d9e0c60` ([run](https://github.com/ai-dynamo/aisimulate/actions/runs/34308649801))
- CodeRabbit reviewed commit: `c730a22c78074d66550cec35f757f7e91d9e0c60`
- Codex reviewed commit: `c730a22c78074d66550cec35f757f7e91d9e0c60`
- Negative or boundary cases: checker tests reject the former malformed Jinja line, wrong license expressions, unterminated block/HTML comments, mismatched delimiters, and dangling closing delimiters
- Expected-value derivation (for exact numerical assertions): N/A
- Before/after output, trace, benchmark, or golden diff: the hardened checker identified exactly 50 malformed templates before remediation and passes all 1,064 tracked source files afterward; all 50 Jinja templates parse successfully and were verified as exact header-only replacements

Additional checks:

- `python3 scripts/check_packaged_legal_files.py` passes and all packaged legal-file copies are byte-identical to their roots
- Ruff and `git diff --check` pass on the exact head
- all seven commits are GitHub-verified SSH signatures and contain DCO `Signed-off-by` trailers
- a fresh merge-tree against `main` at `106134962d15031b149a547d98a4a21e6862fd50` is clean

## Modeling or data provenance

N/A. This PR changes no datasets, performance data, model inputs, or measured results.

## Tracking

- Closes: N/A
- Related PRs or issues: non-dataset OSRB follow-up for AISimulate `0.12.0`

### CODE REVIEW

**Comment by jasonqinzhou** [comment]
@coderabbitai review

File: scripts/check_copyright.py [hunk 1]
```diff
@@ -28,7 +28,12 @@
     ".rule",
     ".sh",
 }
-LICENSE_MARKER = "SPDX-License-Identifier: Apache-2.0"
+LICENSE_MARKER = re.compile(
+    r"^\s*(?:(?:\#|//|/\*+|\*|<!--)\s*)?"
+    r"SPDX-License-Identifier:\s*Apache-2\.0\s*"
+    r"(?:\*/|-->)?\s*$",
```
[the remark this answers is not in the document]
Comment (jasonqinzhou) [reply scripts/check_copyright.py]: Confirmed and fixed in 78aeb2b7. Block and HTML openers now require their matching closing delimiters, while standalone lines and multiline-comment interior lines remain supported. Added rejection coverage for unterminated and mismatched `/*` / `<!--` forms; focused source-compliance validation passes (35 tests).

File: scripts/check_copyright.py [hunk 2]
```diff
@@ -28,7 +28,17 @@
...
 }
-LICENSE_MARKER = "SPDX-License-Identifier: Apache-2.0"
+APACHE_LICENSE_IDENTIFIER = r"SPDX-License-Identifier:\s*Apache-2\.0"
+LICENSE_MARKER = re.compile(
+    rf"^\s*(?:"
+    rf"(?:\#|//)\s*{APACHE_LICENSE_IDENTIFIER}"
+    rf"|/\*+\s*{APACHE_LICENSE_IDENTIFIER}\s*\*/"
+    rf"|<!--\s*{APACHE_LICENSE_IDENTIFIER}\s*-->"
+    rf"|\*\s*{APACHE_LICENSE_IDENTIFIER}(?:\s*\*/)?"
+    rf"|{APACHE_LICENSE_IDENTIFIER}(?:\s*(?:\*/|-->))?"
```
[the remark this answers is not in the document]
Comment (jasonqinzhou) [reply scripts/check_copyright.py]: Confirmed and fixed in c730a22c. The bare-identifier alternative no longer accepts closing-only `*/` or `-->` suffixes; both cases moved from accepted to rejected coverage. Focused source-compliance validation remains 35 passing tests.

**Review by jasonqinzhou — APPROVED** [review]
Reviewed exact head c730a22c78074d66550cec35f757f7e91d9e0c60. Two functional edge cases found during review—unterminated block/HTML comments and dangling closing delimiters—were fixed in 78aeb2b7 and c730a22c. Local validation is green (35 source-compliance tests; 339 generator tests passed with 4 skipped), Fast CI and Full CI are green on the exact head, DCO/codeowners/CodeRabbit pass, all review threads are resolved, and the fresh merge-tree against main at 106134962d15031b149a547d98a4a21e6862fd50 is clean. Approved.
